core/src/test/scala/org/apache/spark/rdd/PipedRDDSuite.scala

```diff
@@ -142,17 +147,23 @@ class PipedRDDSuite extends SparkFunSuite with SharedSparkContext {
     val piped = data.pipe("wc -c")
     assert(piped.count == 8)
     val charCounts = piped.map(_.trim.toInt).collect().toSet
-    assert(Set(0, 4, 5) == charCounts)
+    val expected = if (Utils.isWindows) {
+      // Note that newline character on Windows is \r\n which are two.
+      Set(0, 5, 6)
+    } else {
+      Set(0, 4, 5)
+    }
+    assert(expected == charCounts)
   }
 
   test("pipe with env variable") {
-    if (testCommandAvailable("printenv")) {
+    if (testCommandAvailable(envCommand)) {
       val nums = sc.makeRDD(Array(1, 2, 3, 4), 2)
-      val piped = nums.pipe(Seq("printenv", "MY_TEST_ENV"), Map("MY_TEST_ENV" -> "LALALA"))
+      val piped = nums.pipe(s"$envCommand MY_TEST_ENV", Map("MY_TEST_ENV" -> "LALALA"))
       val c = piped.collect()
-      assert(c.size === 2)
-      assert(c(0) === "LALALA")
-      assert(c(1) === "LALALA")
+      assert(c.length === 2)
+      assert(c(0).contains("LALALA"))
+      assert(c(1).contains("LALALA"))
```

Comment:
This had to be contains because `cmd.exe /C set MY_TEST_ENV` prints

```
MY_TEST_ENV=LALALA
```

whereas `printenv` prints

```
LALALA
```